src/casp/ml/data/distributed_anndata.py

```diff
@@ -185,6 +187,19 @@ def __init__(
         )
         self.var = self.adatas[0].var.copy()
 
+    def __getstate__(self):
+        attributes = self.__dict__.copy()
+        del attributes['cache']
+        del attributes['adatas']        
+        return attributes
+
+    def __setstate__(self, d):
```

Comment:
Is this for pickling? Just curious about the methods.

Reply:
yep -- these are the methods to pickle (__getstate__) and un-pickle (__setstate__)